embark.el

```diff
@@ -2885,6 +2901,18 @@ respects symbol boundaries."
 
 ;;; Setup and pre-action hooks
 
+(defun embark--restart (&rest _)
+  "Restart current command with current input.
+Use this to refresh the list of candidates for commands that do
+not handle that themselves."
+  (when (minibufferp)
+    ;; XXX: This setter is needed in order to allow restarting
+    ;; again after the restart. I don't understand how
+    ;; embark--command is treated in Embark, sometimes it is
+    ;; let-bound, sometimes setq'ed.
```

Comment:
Sometimes using let and sometimes setq is not the result of careful thought and is not on purpose. It's just me being sloppy. Probably they can all be let.